drivers/usb_c/ppc/usbc_ppc_numaker.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2024 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_numaker_ppc
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/usb_c/usbc_ppc.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_numaker.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ppc_numaker, CONFIG_USBC_LOG_LEVEL);
+
+#include <soc.h>
+#include <NuMicro.h>
+
+#include "../tcpc/ucpd_numaker.h"
+
+/* Implementation notes on NuMaker TCPC/PPC/VBUS
+ *
+ * PPC and VBUS rely on TCPC/UTCPD and are just pseudo. They are completely
+ * implemented in TCPC/UTCPD.
+ */
+
+/**
+ * @brief Immutable device context
+ */
+struct numaker_ppc_config {
+	const struct device *tcpc_dev;
+};
+
+/**
+ * @brief Initializes the usb-c ppc driver
+ *
+ * @retval 0 on success
```

Comment:
To be consistent with your following @retval comments, could you add `@retval -ENODEV`

Reply:
added